haystack/nodes/prompt/invocation_layer/anthropic_claude.py

```diff
@@ -0,0 +1,207 @@
+import os
+from typing import Dict, List, Union, Optional
+import json
+import logging
+
+import requests
+import sseclient
+from transformers import GPT2TokenizerFast
+
+from haystack.errors import AnthropicError, AnthropicRateLimitError, AnthropicUnauthorizedError
+from haystack.nodes.prompt.invocation_layer.base import PromptModelInvocationLayer
+from haystack.nodes.prompt.invocation_layer.handlers import TokenStreamingHandler, DefaultTokenStreamingHandler
+from haystack.utils.requests import request_with_retry
+from haystack.environment import HAYSTACK_REMOTE_API_MAX_RETRIES, HAYSTACK_REMOTE_API_TIMEOUT_SEC
+
+ANTHROPIC_TIMEOUT = float(os.environ.get(HAYSTACK_REMOTE_API_TIMEOUT_SEC, 30))
+ANTHROPIC_MAX_RETRIES = int(os.environ.get(HAYSTACK_REMOTE_API_MAX_RETRIES, 5))
+
+logger = logging.getLogger(__name__)
+
+
+class AnthropicClaudeInvocationLayer(PromptModelInvocationLayer):
+    """
+    Anthropic Claude Invocation Layer
+    This layer is used to invoke the Claude API provided by Anthropic.
+    """
+
+    def __init__(self, api_key: str, model_name_or_path: str = "claude-v1", max_length=200, **kwargs):
+        """
+         Creates an instance of PromptModelInvocation Layer for Anthropic's Claude models.
+        :param model_name_or_path: The name or path of the underlying model.
+        :param max_tokens_to_sample: The maximum length of the output text.
+        :param api_key: The Anthropic API key.
+        :param kwargs: Additional keyword arguments passed to the underlying model. The list of Anthropic-relevant
+        kwargs includes: stop_sequences, temperature, top_p, top_k, and stream. For more details about these kwargs,
+        see Anthropic's [documentation](https://console.anthropic.com/docs/api/reference).
+        """
+        super().__init__(model_name_or_path)
+        if not isinstance(api_key, str) or len(api_key) == 0:
+            raise AnthropicError(
+                f"api_key {api_key} must be a valid Anthropic key. Visit https://console.anthropic.com/account/keys to get one."
+            )
+        self.api_key = api_key
+        self.max_length = max_length
+
+        # Due to reflective construction of all invocation layers we might receive some
+        # unknown kwargs, so we need to take only the relevant.
+        # For more details refer to Anthropic documentation
+        supported_kwargs = ["temperature", "top_p", "top_k", "stop_sequences", "stream", "stream_handler"]
+        self.model_input_kwargs = {k: v for (k, v) in kwargs.items() if k in supported_kwargs}
+
+        # Number of max tokens is based on the official Anthropic model pricing from:
+        # https://cdn2.assets-servd.host/anthropic-website/production/images/FINAL-PRICING.pdf
+        self.max_tokens_limit = 9000
+        self.tokenizer: GPT2TokenizerFast = GPT2TokenizerFast.from_pretrained("gpt2")
```

Comment:
We should use their tokenizer. We need to use it somehow as Anthropic does. See [this](https://github.com/anthropics/anthropic-sdk-python/blob/main/anthropic/tokenizer.py)